zookeeper-server/src/main/java/org/apache/zookeeper/server/auth/znode/groupacl/X509ZNodeGroupAclProvider.java

```diff
@@ -136,12 +136,8 @@ public boolean isAuthenticated() {
 
   @Override
   public boolean isValid(String id) {
-    try {
-      new X500Principal(id);
-      return true;
-    } catch (IllegalArgumentException e) {
-      return false;
-    }
+    // Id can be of multiple format since it can be either a domain or a client URI
```

Comment:
I was wondering could we do following things here : 
1. Check if id is of supported clientURI format
2. Check if id is of registered domains. (have a list of domains in memory)
I'm not completely convinced by above suggestion but just wanted to check what do you think about it. Just make sure that it's not loop hole in security.

Reply:
In X509GroupZNodeACL case, this value is either domain or clientURI. The only values not controlled by us is set by superusers, who we should trust. Therefore it doesn't seem necessary to perform this check.